modules/fundamental/src/vulnerability/service/mod.rs

```diff
@@ -30,17 +36,99 @@ impl VulnerabilityService {
     ) -> Result<PaginatedResults<VulnerabilitySummary>, Error> {
         let connection = self.db.connection(&tx);
 
+        let inner_query = vulnerability::Entity::find()
+            .left_join(cvss3::Entity)
+            .expr_as_(
+                SimpleExpr::FunctionCall(Func::avg(SimpleExpr::Column(
+                    trustify_entity::cvss3::Column::Score.into_column_ref(),
+                ))),
+                "average_score",
+            )
+            .expr_as_(
+                SimpleExpr::FunctionCall(Func::cust("cvss3_severity".into_identity()).arg(
+                    SimpleExpr::FunctionCall(Func::avg(SimpleExpr::Column(
+                        trustify_entity::cvss3::Column::Score.into_column_ref(),
+                    ))),
+                )),
+                "average_severity",
+            )
+            .group_by(vulnerability::Column::Id);
+
+        let mut outer_query = vulnerability::Entity::find();
+
+        // Alias the inner query as exactly the table the entity is expecting
+        // so that column aliases link up correctly.
+        QueryTrait::query(&mut outer_query)
+            .from_clear()
+            .from_subquery(inner_query.into_query(), "vulnerability".into_identity());
+
+        /*
         let limiter = vulnerability::Entity::find().filtering(search)?.limiting(
             &connection,
             paginated.offset,
             paginated.limit,
         );
 
+         */
+
+        let limiter = outer_query
+            .column_as(
+                SimpleExpr::Column(ColumnRef::Column(
+                    "average_score".into_identity().into_iden(),
+                )),
+                "average_score",
+            )
+            .column_as(
+                SimpleExpr::Column(ColumnRef::Column(
+                    "average_severity".into_identity().into_iden(),
+                ))
+                .cast_as("TEXT".into_identity()),
+                "average_severity",
+            )
+            .filtering_with(
+                search,
+                Columns::from_entity::<advisory::Entity>()
```

Comment:
@bobmcwhirter When I try a free text search, e.g. `q=CVE-2024-36974`, which should match the `identifier` field, I get an error from the db: missing FROM-clause entry for table "advisory". We're getting columns from `advisory::Entity` but we're querying `vulnerability::Entity`. I'm guessing we need a join of some type, but all my attempts have failed. Help?

Reply:
Ah, I think I just bad copy-pasta'd `advisory::Entity` and it should've been `vulnerability::Entiy` for the `Columns::from_entity`

Didja give that a whirl?